pc-relate/pcrelate.wdl

```diff
@@ -0,0 +1,395 @@
+
+		f.write('out_file_all "~{out_prefix_final}_all.pdf"\n')
+		f.write('out_file_cross "~{out_prefix_final}_cross_group.pdf"\n')
+		f.write('out_file_study "~{out_prefix_final}_within_group.pdf"\n')
+
+		if "~{phenotype_file}" is not "":
+			f.write('phenotype_file "~{phenotype_file}"\n')
+		
+		if "~{group}" is not "":
+			f.write('study "~{group}"\n')
+
+		if "~{sample_include_file}" is not "":
+			f.write('sample_include_file "~{sample_include_file}"\n')
+
+		CODE
+
+		R -q --vanilla --args kinship_plots.config < /usr/local/analysis_pipeline/R/kinship_plots.R
+	}
+	
+	# Estimate disk size required
+	Int kin_size = ceil(size(kinship_file, "GB"))
+	Int phen_size = select_first([ceil(size(phenotype_file, "GB")), 0])
+	Int smpl_size = select_first([ceil(size(sample_include_file, "GB")), 0])
+	Int finalDiskSize = kin_size + phen_size + smpl_size + addldisk
+	
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master@sha256:0bb7f98d6b9182d4e4a6b82c98c04a244d766707875ddfd8a48005a9f5c5481e"
+		disks: "local-disk " + finalDiskSize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		Array[File] kinship_plots = glob("*.pdf")
+	}
+}
+
+workflow pcrel {
+	input {
+		File pca_file
+		File gds_file
+		File? sample_include_file
+		Int? n_pcs # sb visual default is 3, check if gets passed in as such
+		File? variant_include_file
+		Int? variant_block_size
+		String? out_prefix
+		Int? n_sample_blocks # sb default 1
+		File? phenotype_file
+		Float? kinship_plot_threshold
+		String? group
+		Float? sparse_threshold
+		Boolean ibd_probs = true
+	}
+
+	call pcrelate_beta {
+		input:
+			gds_file = gds_file,
+			pca_file = pca_file,
+			n_pcs = n_pcs,
+			out_prefix = out_prefix,
+			sample_include_file = sample_include_file,
+			variant_include_file = variant_include_file,
+			variant_block_size = variant_block_size
+	}
+
+	call sample_blocks_to_segments {
+		input:
+			n_sample_blocks = n_sample_blocks
+	}
+
+	scatter(seg in sample_blocks_to_segments.segments) {
+		call pcrelate {
+			input:
+				gds_file = gds_file,
+				pca_file = pca_file,
+				beta_file =pcrelate_beta.beta,
+				n_pcs = n_pcs,
+				out_prefix = out_prefix,
+				variant_include_file = variant_include_file,
+				variant_block_size = variant_block_size,
+				sample_include_file = sample_include_file,
+				n_sample_blocks = n_sample_blocks,
+				segment = seg,
+				ibd_probs = ibd_probs
+		}
+	}
+
+	call pcrelate_correct {
+		input:
+			n_sample_blocks = n_sample_blocks,
+			pcrelate_block_files = pcrelate.block,
+			sparse_threshold = sparse_threshold
+	}
+
+	if (ibd_probs) { # this matches the run_plots = ibd_probs part of the CWL
+		call kinship_plots {
+			input:
+				kinship_file = pcrelate_correct.pcrelate_output,
+				kinship_plot_threshold = kinship_plot_threshold,
+				phenotype_file = phenotype_file,
+				group = group,
+				sample_include_file = sample_include_file,
+				out_prefix_initial = out_prefix
```

Comment:
It looks like you aren't passing kinship_method anywhere

Reply:
This is hardcoded in the CWL, but it isn't possible to perfectly replicate how it's hardcoded in WDL so I had to get a little creative.

The parent CWL does not have an input for kinship method, but it does pass this value in to the plot task. It can do this because although kinship_method lacks a source, [it does have a default](https://github.com/UW-GAC/analysis_pipeline_cwl/blob/8d9855ce949330d726cbc26341aaa1854089e02e/relatedness/pc-relate-wf.cwl#L259). Therefore it passes in a hardcoded default to the task.

We can verify this in Seven Bridges. The plots task lacks kinship_method as a valid input the user can set before running the workflow. 
<img width="358" alt="Screenshot 2021-09-22 at 1 21 59 PM" src="https://user-images.githubusercontent.com/27784612/134416454-31280ebe-ee90-4db9-8856-1a8850f69212.png">

But after running the workflow, it shows up as a grayed-out input whose value matches the hardcoded input.
<img width="395" alt="Screenshot 2021-09-22 at 1 21 53 PM" src="https://user-images.githubusercontent.com/27784612/134416491-127f7abd-c1b9-436e-94b8-708b1d9eb58b.png">

Since the user cannot change this at all in SB, it's actually more appropriate for the WDL to take this as a private input rather than the public input with a default, so I'm glad you brought this up. I moved the variable to make it private.

Reply:
Similar to the out_prefix discussion above. We should mirror the task-level to the tool CWL here. So we should pass the (hardcoded) kinship_method in the call.